plutus-core/untyped-plutus-core/src/UntypedPlutusCore/Transform/Inline.hs

```diff
@@ -470,7 +470,8 @@ fullyApplyAndBetaReduce info args0 = do
                   acc
               go acc' params args'
             else pure Nothing
-        _ -> pure Nothing
+        -- no more arguments to apply, just apply what we have
+        (_, []) -> pure $ Just acc
```

Comment:
This also needs to be changed

Reply:
Yes I just wanted to see if the PIR part was right first.